services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -17,7 +17,15 @@ class ProjectService(projectRepository: ProjectRepository, projectVersioning: Pr
     projectRepository.getProjectById(projectId).flatMap(getForUserOrFail(_, userId))
 
   def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
-    projectRepository.getProjectByName(namePattern).flatMap(getForUserOrFail(_, userId))
+    projectRepository
+      .getProjectsByName(namePattern)
+      .flatMap(
+        projects =>
+          projects.find(_.ownerId == userId).orElse(projects.headOption) match {
+            case Some(project) => Future.successful(project)
+            case None          => Future.failed(new ProjectNotFoundException)
+          }
+      )
 
   private def getForUserOrFail(project: Option[Project], userId: UserId): Future[Project] =
     project match {
```

Comment:
If we don`t use getForUserOrFail, we should to remowe this method. I think.

Reply:
This method (getForUserOrFail) are used in other methods, therefore i leave it.